### PR TITLE
Update static file docs and log if WebRootPath is not found 

### LINKED ISSUE #38481: `UseStaticFiles` should display the default PhysicalFileProvider default path when using a constructor that doesn't override the default `IFileProvider`
### Is your feature request related to a problem? Please describe.

This request is related to a discoverability issue for `UseStaticFiles` extension. If do not start from a default template and add `UseStaticFiles()`, the inline xmldoc do not describe the default local physical path it will try to fetch the static files from. 

* Running `dotnet run` does not log the issue in the console that no path `/wwwroot` exist in the project and that `UseStaticFiles` will do nothing.
* Creating a static file at the root and trying to match the url and the physical path only returns a 404 without any more help.

image

### Describe the solution you'd like

Default values like `/wwwroot` for static files should be document in the xmldoc of the API at the point of calling as to help people avoid having to switch context between their code and the docs. 

Additionally, calling the extension `UseStaticFiles` while the default physical path `/wwwroot` doesn't exist should create an warning message in the logs to warn the developer of a misconfiguration. If implemented, this message should only be shown once as to avoid polluting the logs.

### Additional context

My context here is for new adopter of .NET that are simply trying to create a simple API. It should be easy for them to create code without relying on the docs unless they are stuck.

### PR DESCRIPTION
**Pull request by Tratcher**
Fixes #38481

AspNetCore only serves static files from the wwwroot subfolder by default. This is different from ASP.NET Framework that served from the root directory by default.

Currently there's no warning or error if the wwwroot directory does not exist.

This adds xml docs to clarify the default, and logs warning if the wwwroot directory is missing and you have a StaticFile middleware.

### CODE REVIEW

File: src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs [hunk 1]
```diff
@@ -158,6 +159,12 @@ static string ExpandPorts(string ports, string scheme)
         await Server.StartAsync(httpApplication, cancellationToken);
         HostingEventSource.Log.ServerReady();
 
+        // See HostingEnvironmentExtensions.Initialize
+        if (HostingEnvironment.WebRootFileProvider is NullFileProvider)
+        {
+            Log.WebRootPathNotFound(LifetimeLogger, Path.GetFullPath(Path.Combine(HostingEnvironment.ContentRootPath, HostingEnvironment.WebRootPath ?? "wwwroot")));
+        }
```
Comment (davidfowl) [inline src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs:166]: Doesn't this always end up logging if you don't have a web root? We don't want that.

**Comment by davidfowl** [comment]
> TODO: Should this only log from the StaticFileMiddleware constructor instead?

Yes.

File: src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs (hunk 1 above)
Comment (Tratcher) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs]: Moved to the StaticFileMiddleware. Now it only logs if you don't have webroot but do have UseStaticFile.

File: src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs [hunk 2]
```diff
@@ -17,6 +18,8 @@ public static class DefaultFilesExtensions
     /// </summary>
     /// <param name="app"></param>
     /// <returns></returns>
+    /// <remarks>This defaults to serving files from the path specified in <see cref="IWebHostEnvironment.WebRootPath"/>
+    /// which defaults to the 'wwwroot' subfolder.</remarks>
```
Comment (JamesNK) [inline src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs:22]:
```suggestion
    /// <remarks>
    /// This defaults to serving files from the path specified in <see cref="IWebHostEnvironment.WebRootPath"/>
    /// which defaults to the 'wwwroot' subfolder.
    /// </remarks>
```
nit

File: src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs [hunk 3]
```diff
@@ -33,6 +36,8 @@ public static IApplicationBuilder UseDefaultFiles(this IApplicationBuilder app)
     /// <param name="app"></param>
     /// <param name="requestPath">The relative request path.</param>
     /// <returns></returns>
+    /// <remarks>This defaults to serving files from the path specified in <see cref="IWebHostEnvironment.WebRootPath"/>
+    /// which defaults to the 'wwwroot' subfolder.</remarks>
```
Comment (JamesNK) [inline src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs:40]:
```suggestion
    /// <remarks>
    /// This defaults to serving files from the path specified in <see cref="IWebHostEnvironment.WebRootPath"/>
    /// which defaults to the 'wwwroot' subfolder.
    /// </remarks>
```

File: src/Middleware/StaticFiles/src/DirectoryBrowserExtensions.cs [hunk 4]
```diff
@@ -17,6 +18,8 @@ public static class DirectoryBrowserExtensions
     /// </summary>
     /// <param name="app"></param>
     /// <returns></returns>
+    /// <remarks>This defaults to browsing files from the path specified in <see cref="IWebHostEnvironment.WebRootPath"/>
+    /// which defaults to the 'wwwroot' subfolder.</remarks>
```
Comment (JamesNK) [inline src/Middleware/StaticFiles/src/DirectoryBrowserExtensions.cs:22]: Repeat for all: multiline XML elements should have element start/end on separate lines.

File: src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs (hunk 2 above)
Comment (JamesNK) [reply src/Middleware/StaticFiles/src/DefaultFilesExtensions.cs:22]: language nit:

I'm not a fan of two defaults in one sentence.

How about:

```
Files are served from the path specified by <see cref="IWebHostEnvironment.WebRootPath"/>.
The default path is the 'wwwroot' subfolder.
```

File: src/Middleware/StaticFiles/src/LoggerExtensions.cs [hunk 5]
```diff
@@ -58,4 +58,8 @@ public static void FileServed(this ILogger logger, string virtualPath, string ph
 
     [LoggerMessage(14, LogLevel.Debug, "The file transmission was cancelled", EventName = "WriteCancelled")]
     public static partial void WriteCancelled(this ILogger logger, Exception ex);
+
+    [LoggerMessage(16, LogLevel.Warning,
+        "The WebRootPath was not found: {webRootPath}. Static files may be unavailable.", EventName = "WebRootPathNotFound")]
```
Comment (JamesNK) [inline src/Middleware/StaticFiles/src/LoggerExtensions.cs:63]:
```suggestion
        "The WebRootPath was not found: {WebRootPath}. Static files may be unavailable.", EventName = "WebRootPathNotFound")]
```
These names in logs are pascal case.

**Review by JamesNK — CHANGES_REQUESTED** [review]
Unit test log is written?

File: src/Middleware/StaticFiles/src/LoggerExtensions.cs (hunk 5 above)
Comment (Tratcher) [reply src/Middleware/StaticFiles/src/LoggerExtensions.cs:63]: Does that cause some different behavior? Where are these accessible from?

**Review by JamesNK — APPROVED** [review]

**Pull request merged**